optuna/study.py

```diff
@@ -72,11 +95,11 @@ def best_trial(self) -> FrozenTrial:
         return copy.deepcopy(self._storage.get_best_trial(self._study_id))
 
     @property
-    def direction(self) -> StudyDirection:
+    def direction(self) -> Union[StudyDirection, Sequence[StudyDirection]]:
```

Comment:
Curious to hear what you think about providing an entirely new API for plural directions and such (incl. `value`). It might be clearer to actually deprecate these existing APIs and always return sequences (tuples).

Reply:
I think we should do that since it leads to simple source codes. Possible naming conventions include the following.
- `direction` and `directions`
- `value` and `values`
- `intermediate_values` and `intermediate_values_list`

What do you think?